### PR TITLE
Even even more colums

### PR DESCRIPTION
**Pull request by xwatkins**

### CODE REVIEW

File: src/view/uniprotkb/components/__tests__/__snapshots__/CatalyticActivityView.spec.tsx.snap [hunk 1]
```diff
@@ -2,9 +2,6 @@
 
 exports[`CatalyticActivityView component should render catalytic activity 1`] = `
 <DocumentFragment>
-  <h4>
```
Comment (dlrice) [inline src/view/uniprotkb/components/__tests__/__snapshots__/CatalyticActivityView.spec.tsx.snap]: the props have changed for this component and now needs the title provided

File: src/view/uniprotkb/components/__tests__/__snapshots__/DiseaseInvolvementView.spec.tsx.snap [hunk 2]
```diff
@@ -2,9 +2,6 @@
 
 exports[`DiseaseInvolvement should render DiseaseInvolvement 1`] = `
 <DocumentFragment>
-  <h3>
```
Comment (dlrice) [inline src/view/uniprotkb/components/__tests__/__snapshots__/DiseaseInvolvementView.spec.tsx.snap]: the props have changed for this component and now needs the title provided

File: src/view/uniprotkb/components/VariationView.tsx [hunk 3]
```diff
@@ -175,20 +177,22 @@ const VariationView: FC<{ primaryAccession: string }> = ({
...
-            length={data.sequence.length}
-            sequence={data.sequence}
-            height="20"
-          />
-          <protvista-filter />
-          <protvista-variation length={data.sequence.length}>
-            <protvista-variation-adapter ref={setTrackData} />
-          </protvista-variation>
-        </div>
+        {!noTable && (
```
Comment (dlrice) [inline src/view/uniprotkb/components/VariationView.tsx]: i'm being pedantic here but am wondering if something like `showTable` would be slightly easier to mentally compute as we're only taking the logical negation of `noTable`?

File: src/view/uniprotkb/components/ProteinNamesView.tsx [hunk 4]
```diff
@@ -81,6 +81,22 @@ const ProteinDescriptionView: React.FC<{
...
+  ecNumbers: ValueWithEvidence[];
+  isCompact?: boolean;
+}> = ({ ecNumbers, isCompact = false }) => (
+  <Fragment>
+    {ecNumbers.map(
+      (ecNumber): JSX.Element =>
+        isCompact ? (
+          <Fragment key={v1()}>{ecNumber.value}</Fragment>
+        ) : (
+          <NameWithEvidence data={ecNumber} key={v1()} />
```
Comment (dlrice) [inline src/view/uniprotkb/components/ProteinNamesView.tsx]: No action needed now but suggesting it as a larger conversation: do want to add uuid when we convert the data (across the whole front-end codebase)? This way the ids will be static and the tree diff'ing can work as normal.

**Comment by dlrice** [comment]
Seeing `ColumnConfiguration.tsx:314 Uncaught TypeError: Cannot read property 'value' of undefined` when attempting to access `sequenceData.sequence.value`:
```
ColumnConfiguration.set(Column.sequence, {
  label: 'Sequence',
  render: data => {
    const sequenceData = data[EntrySection.Sequence];
    return (
      <Sequence
        sequence={sequenceData.sequence.value}
        id={data.primaryAccession}
      />
    );
  },
});
```

**Comment by dlrice** [comment]
> Seeing `ColumnConfiguration.tsx:314 Uncaught TypeError: Cannot read property 'value' of undefined` when attempting to access `sequenceData.sequence.value`:
> 
> ```
> ColumnConfiguration.set(Column.sequence, {
>   label: 'Sequence',
>   render: data => {
>     const sequenceData = data[EntrySection.Sequence];
>     return (
>       <Sequence
>         sequence={sequenceData.sequence.value}
>         id={data.primaryAccession}
>       />
>     );
>   },
> });
> ```

ok, which disappears when i reload the page. hmm. not sure about this.

**Comment by dlrice** [comment]
I noticed that the `Position(s) | Change | Description | Disease association` table is displayed even when there are table rows:

Screenshot 2019-12-17 at 12 37 01

File: src/model/utils/utils.ts [hunk 5]
```diff
@@ -6,6 +6,9 @@ const hasContent = (obj: any) => {
       return valArray.length > 0;
     }
     if (typeof val === 'object' && val) {
+      if (val instanceof Map) {
+        return Array.from(val.entries()).some(entry => entry.length > 0);
```
[the remark this answers is not in the document]
Comment (xwatkins) [reply src/model/utils/utils.ts]: Updated and wrote test cases for this util 👍

File: src/view/uniprotkb/FunctionSection.tsx [hunk 6]
```diff
@@ -122,6 +149,13 @@ const FunctionSection: FC<{
               CommentType.CATALYTIC_ACTIVITY
             ) as CatalyticActivityComment[]
           }
+          title={CommentType.CATALYTIC_ACTIVITY.toLocaleLowerCase()}
+        />
+        <CofactorView
+          cofactors={
+            data.commentsData.get(CommentType.COFACTOR) as CofactorComment[]
+          }
+          title={CommentType.COFACTOR.toLocaleLowerCase()}
```
[the remark this answers is not in the document]
Comment (xwatkins) [reply src/view/uniprotkb/FunctionSection.tsx]: Typo!

File: src/view/uniprotkb/components/VariationView.tsx (hunk 3 above)
Comment (xwatkins) [reply src/view/uniprotkb/components/VariationView.tsx]: 👍

File: src/view/uniprotkb/components/ProteinNamesView.tsx (hunk 4 above)
Comment (xwatkins) [reply src/view/uniprotkb/components/ProteinNamesView.tsx]: Most definitely as this is currently useless. Will make a Jira

**Comment by xwatkins** [comment]
Comments addressed. Updated code so empty table of variants not displayed if no variants returned.

**Pull request merged**